src/lib/components/SparklineChart/index.js

```diff
@@ -39,187 +41,121 @@ function clampTo2Decimals(val) {
-
-    /** @private {!Array<!LighthouseScore>} */
-    this.medians_ = [];
-
-    /** @private {number} */
-    this.stroke_ = 2;
-    /** @private {number} */
-    this.circleRadius_ = 3;
-    /** @private {number} */
-    this.padding_ = 0;
-    /** @private {number} */
-    this.scoreHeight_ = 15;
-    /** @private {number} */
-    this.transformY_ = this.scoreHeight_ + 30;
-
-    /** @private {?number} */
-    this.width_ = null;
-    /** @private {?number} */
-    this.height_ = null;
-
-    /** @private {?HTMLElement} */
-    this.cursor_ = null;
-    /** @private {?HTMLElement} */
-    this.score_ = null;
-    /** @private {?HTMLElement} */
-    this.scoreValueText_ = null;
-    /** @private {?HTMLElement} */
-    this.scoreDateText_ = null;
-    /** @private {?HTMLElement} */
-    this.announcer_ = null;
-
-    /** @private {?DataPoint} */
-    this.point_ = null;
+    this.topPadding = 20;
+    this.circleRadius = 3;
+    this.strokeWidth = 2;
 
-    /** @export {!Array<!DataPoint>} */
-    this.datapoints = [];
-  }
+    // private, updated in render() by calling getBoundingClientRect
+    this.width_ = 0;
+    this.height_ = 0;
 
-  /** @export @override */
-  static get observedAttributes() {
-    return ["fill", "showlast"];
-  }
+    // cursor selection, causes non-render() changes
+    this.point_ = null;
 
-  /**
-   * @return {!Array<!LighthouseScore>}
-   * @export
-   */
-  get values() {
-    return this.values_;
-  }
+    // elements stolen from renderRoot
+    this.cursorElement_ = null;
+    this.scoreElement_ = null;
+    this.announcerElement_ = null;
+    this.scoreValueText_ = null;
+    this.scoreDateText_ = null;
 
-  /**
-   * @param {!Array<!LighthouseScore>} val
-   * @export
-   */
-  set values(val) {
-    this.values_ = val;
-    this.update_();
-  }
+    // processed points from values that allow mouse focus
+    this.datapoints = [];
 
-  /**
-   * @return {?Promise<!Array<!LighthouseScore>>} val
-   * @export
-   */
-  get medians() {
-    return this.mediansPromise_;
-  }
+    // bind so this can be added/removed from a global handler
+    this.onResize = this.onResize.bind(this);
 
-  /**
-   * @param {?Promise<!Array<!LighthouseScore>>} val
-   * @export
-   */
-  set medians(val) {
-    // nb. Because `Promise.resolve` is used, this could accept a non-Promise.
-    const localPromise = Promise.resolve(val || []);
-    this.mediansPromise_ = localPromise;
-
-    localPromise.then((medians) => {
-      if (localPromise !== this.mediansPromise_) {
-        return; // something changed
-      }
-      this.medians_ = medians;
-      this.update_();
-    });
+    if (HAS_RESIZE_OBSERVER) {
+      // We rely on our user ensuring that <web-sparkline-chart> has size.
+      const ro = new ResizeObserver(this.onResize);
+      ro.observe(this);
+    }
   }
 
-  /**
-   * @return {boolean}
-   * @export
-   */
-  get fill() {
-    return this.hasAttribute("fill");
-  }
+  set demo(demo) {
```

Comment:
Do you want to keep this or remove it before we land?

Reply:
I'm less worried since this isn't hundreds of kbs of data.